### PR TITLE
docs: cabal repl fresh-up

### PR DESCRIPTION
**Pull request by ulysses4ever**
Didn't do any of the listed below, sorry.

---
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [x] The documentation has been updated, if necessary.
* [ ] Include [manual QA notes](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#qa-notes) if your PR relates to cabal-install.

Bonus points for added automated tests!

### CODE REVIEW

File: doc/cabal-commands.rst [hunk 1]
```diff
@@ -791,49 +791,72 @@ it may invoke the configuration step (see ``cabal configure``).
...
-specify these options to the invoked repl. (This flag also works on ``cabal
-repl`` and ``Setup repl`` on sufficiently new versions of Cabal.)
-
-The ``repl-no-load`` flag disables the loading of target modules at startup.
+``cabal repl TARGET [FLAGS]``
+opens an interactive session for the target component within the project and
+loads all of the modules of the target into GHCi as interpreted bytecode.
+The available
+targets are the same as for the ``build`` command: individual components
+within packages in the project, including libraries, executables, test-suites
```
Comment (ffaf1) [inline doc/cabal-commands.rst:799]: “or benchmarks” -> “and benchmarks”?

File: doc/cabal-commands.rst [hunk 2]
```diff
@@ -791,49 +791,72 @@ it may invoke the configuration step (see ``cabal configure``).
...
+or benchmarks. Local packages can also be specified in which case the library
```
Comment (ffaf1) [inline doc/cabal-commands.rst:800]: “Local packages can also be specified in which case…” maybe semicolon

“Local packages can also be specified; in this case…”

File: doc/cabal-commands.rst [hunk 3]
```diff
@@ -791,49 +791,72 @@ it may invoke the configuration step (see ``cabal configure``).
...
+Configuration flags can be specified on the command line and these extend the project
+configuration from the 'cabal.project', 'cabal.project.local' and other files.
 
-    $ cabal repl --build-depends "vector >= 0.12 && < 0.13"
+.. option:: --repl-options
 
-Both of these commands do the same thing as the above, but only exposes ``base``,
-``vector``, and the ``vector`` package's transitive dependencies even if the user
-is in a project context.
+    To avoid ``ghci`` specific flags from triggering unneeded global rebuilds these
```
Comment (ffaf1) [inline doc/cabal-commands.rst:823]: maybe comma after “global rebuilds”

File: doc/cabal-commands.rst [hunk 4]
```diff
@@ -791,49 +791,72 @@ it may invoke the configuration step (see ``cabal configure``).
...
+    flags are stripped from the internal configuration. As a result,
+    ``--ghc-options`` will no longer (reliably) work to pass flags to ``ghci`` (or
+    other repls). Instead, you should use the ``--repl-options`` flag to
```
Comment (ffaf1) [inline doc/cabal-commands.rst:826]: REPLs for consistency with the chunk below.

File: doc/cabal-commands.rst [hunk 5]
```diff
@@ -791,49 +791,72 @@ it may invoke the configuration step (see ``cabal configure``).
...
+    them manually or to install them globally.
+
+    This command opens a REPL with the current default target loaded, and a version
+    of the ``vector`` package matching that specification exposed.
+
+    ::
+
+        $ cabal repl --build-depends "vector >= 0.12 && < 0.13"
+
+    Both of these commands do the same thing as the above, but only exposes ``base``,
```
Comment (ffaf1) [inline doc/cabal-commands.rst:845]: exposes -> expose

**Review by ffaf1 — APPROVED** [review]
^_^d

**Review by Mikolaj — APPROVED** [review]
Whitespace died, but local build of the User's Guide went fine. I forgot, what's the address of the cabal.readthedocs.io build that we could have a look at? Or am I hallucinating there being one?

File: doc/cabal-commands.rst (hunk 2 above)
Comment (ulysses4ever) [reply doc/cabal-commands.rst:800]: Comma should work, I hope.

File: doc/cabal-commands.rst (hunk 1 above)
Comment (ulysses4ever) [reply doc/cabal-commands.rst:799]: Done.

File: doc/cabal-commands.rst (hunk 3 above)
Comment (ulysses4ever) [reply doc/cabal-commands.rst]: Done.

File: doc/cabal-commands.rst (hunk 4 above)
Comment (ulysses4ever) [reply doc/cabal-commands.rst]: Done.

File: doc/cabal-commands.rst (hunk 5 above)
Comment (ulysses4ever) [reply doc/cabal-commands.rst]: Done.

**Comment by ulysses4ever** [comment]
Correction, the actual link to the rendered version: https://cabal--8907.org.readthedocs.build/en/8907/cabal-commands.html#cabal-repl

**Comment by ulysses4ever** [comment]
@Mergify backport 3.10

**Comment by fgaz** [comment]
@ulysses4ever remember to delete your branches after the merge, or prefix branch names with your nickname like @gbaz does, or use branches from your own fork, so we don't end up with branches we don't know the purpose of and whether to delete them

**Comment by ulysses4ever** [comment]
@fgaz good point, thank you. There is some setting about auto removal, I thought? Maybe we should just find and turn it on...

**Comment by ulysses4ever** [comment]
I went ahead and checked the auto-removal box as described here https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-the-automatic-deletion-of-branches

**Pull request merged**